generated/1.28/crds/config.supervisor.pinniped.dev_federationdomains.yaml

```diff
@@ -116,6 +115,7 @@ spec:
                       - kind
                       - name
                       type: object
+                      x-kubernetes-map-type: atomic
```

Comment:
What does this do? Are we okay with this change?

Reply:
According to https://kubernetes.io/docs/reference/using-api/server-side-apply/,

> Applicable to maps. atomic means that the map can only be entirely replaced by a single manager. granular means that the map supports separate managers updating individual fields.

Reply:
I think that this only applies to `managedFields`, which does not appear to be used in Pinniped.

Reply:
Looking at this more closely, `x-kubernetes-map-type: atomic` got applied to:
- `objectRef`, which is of type `corev1.TypedLocalObjectReference`
- `jwks`, `tokenSigningKey`, `stateSigningKey`, and `stateEncryptionKey` which are all of type `corev1.LocalObjectReference`

So in all these cases the `+structType=atomic` tag is coming from the author's of the `corev1` package, so that seems fine. Presumably they know what they are doing with these server-side apply settings. :)

Reply:
Just to test the upgrade path, I tried applying this CRD from `main` and then applying this CRD from this PR branch. It applied without complaint about adding `x-kubernetes-map-type` to these fields. It did not seem to have any impact on my existing FederationDomain resource either. Seems like it worked fine to upgrade the CRD to add these settings.